tools/ci_build/github/apple/package_assembly_utils.py

```diff
@@ -0,0 +1,173 @@
+    exclude_files = [
+        str(path.relative_to(src_root)) for pattern in excluded_file_patterns for path in src_root.glob(pattern)
+    ]
+
+    # return the difference
+    return list(set(all_files) - set(exclude_files))
+
+
+def copy_repo_relative_to_dir(subpath: str, patterns: List[str], dest_dir: pathlib.Path):
+    """
+    Copies file paths relative to the repo root to a directory.
+    The given paths or path patterns are relative to the repo root, and the
+    repo root-relative intermediate directory structure is maintained.
+
+    :param patterns The paths or path patterns relative to the repo root.
+    :param dest_dir The destination directory.
+    """
+    if subpath:
+        src_root = repo_root /  subpath
+    else:
+        src_root = repo_root
+    paths = [path for pattern in patterns for path in src_root.glob(pattern)]
+    for path in paths:
+        repo_relative_path = path.relative_to(src_root)
+        dst_path = dest_dir / repo_relative_path
+        os.makedirs(dst_path.parent, exist_ok=True)
+        if os.path.isdir(path):
+            shutil.copytree(path, dst_path)
+        else:
+            shutil.copy(path, dst_path)
+
+
+def load_json_config(json_config_file: pathlib.Path):
+    """
+    Loads configuration info from a JSON file.
+
+    :param json_config_file The JSON configuration file path.
+    :return The configuration info values.
+    """
+    with open(json_config_file) as config:
+        return json.load(config)
+
+
+def get_podspec_values(framework_info):
+    """
+    Get the podspec deployement targets and weak framework info from the dictionary that load_json_config returned.
+    Looks for iphonesimulator, iphoneos and macos settings.
+    Handles missing platforms and checks consistency.
+    Returns empty string for deployment target if that platofrm is not enabled.
+
+    :return (ios_deployment_target, macos_deployment_target, weak_framework)
+    """
+    ios_deployment_target = ""
+    macos_deployment_target = ""
+    weak_framework = ""  # should be the same for all platforms
+    # get info, allowing for a subset of platforms to be specified
+    for framework in ("iphonesimulator", "iphoneos", "macosx"):
+        if framework not in framework_info:
+            continue
+
+        target = framework_info[framework]["APPLE_DEPLOYMENT_TARGET"]
+        weak = framework_info[framework]["WEAK_FRAMEWORK"]
+
+        if not weak_framework:
+            weak_framework = weak
+        else:
+            # should be consistent
+            assert weak == weak_framework
+
+        if framework == "macosx":
+            macos_deployment_target = target
+        else:
+            if not ios_deployment_target:
+                ios_deployment_target = target
+            else:
+                # should be consistent
+                assert ios_deployment_target == target
+
+    return (ios_deployment_target, macos_deployment_target, weak_framework)
+
+
+def get_ort_version():
```

Comment:
nit: maybe name it `get_ort_genai_version`?

Reply:
done